V2 : "Get the code" button fixed. Issue solettaproject/soletta-dev-app#62

kamidipreetham opened the pull request:
Changes since V1:
- Removed ExamplesView.js.
- Removed all ExamplesView.js references.
- Corrected indentation and spaces in code.
- Added error message when loading file from local source in Cheat Sheet.

[V1](https://github.com/solettaproject/soletta/pull/1835):
-Corrected copy code functionality using execCommand API.
-Moved all of ExamplesView.js code inline as execCommand API doesn't work in injected scripts.
Signed-off-by: Kamidi Preetham <kamidi@live.com>

brunobottazzini commented:
@kamidipreetham, Your commit got a little bit messy. I think it is because of the squash.

Try rewriting and adding the signed off in the last line, where it used to be.

brunobottazzini commented:
Also your patch is outdated with the master and has a conflicts to it.

Rebase it with the soletta master.

brunobottazzini commented on doc/node-types-html/index.html.in:
```diff
@@ -15,10 +15,225 @@
+            var y = /^file?:/i;
+            if(y){
+                alert("Error: Running cheat-sheet from local source! \nPlease run cheat-sheet from server");
+            }
```
This error message shouldn't be here. Let's warn the user about this error when he clicks in the copy button

brunobottazzini commented on doc/node-types-html/index.html.in:
```diff
@@ -15,10 +15,225 @@
+            info.append(title);
+            info.append(description);
+            // Check for local load file requests
+            var x = location.protocol;
```
Is this var used ?

brunobottazzini commented on doc/node-types-html/index.html.in:
```diff
@@ -15,10 +15,225 @@
+        function copyExampleCode(event){
+            var x = document.getElementById('example-info-code');
+            selecttext(x);
+            y=copycode();
```
Split the returned var from the function. 

```y =  copycode();```

Also where do you declare this var ? Why not using it to verify if the copycode was successful ?

brunobottazzini commented on doc/node-types-html/index.html.in:
```diff
@@ -15,10 +15,225 @@
+            var x = document.getElementById('example-info-code');
+            selecttext(x);
+            y=copycode();
+            console.log(y);
```
We don't need this console.log here

brunobottazzini commented on doc/node-types-html/index.html.in:
```diff
@@ -15,10 +15,225 @@
+                hidden_code.html(String(data));
+            }); 
+            getCodeButton.on('click',copyExampleCode);
+            }
```
Indent all this code inside the else

kamidipreetham commented:
V3:
- Error message now pops up when user clicks "Get the code" button if user loads Cheat Sheet from local source.
- Removed unnecessary console.log()
- Added proper signed off message.
